agent/connect/ca/provider.go

```diff
@@ -74,3 +75,10 @@ type Provider interface {
 	// created for an intermediate CA.
 	Cleanup() error
 }
+
+// NeedsLogger is an optional interface that allows a CA provider to use the
+// Consul logger to output diagnostic messages.
+type NeedsLogger interface {
+	// SetLogger will pass a configured Logger to the provider.
+	SetLogger(logger *log.Logger)
```

Comment:
@s-christoff @banks Should this start out as a `hclog.Logger` since it's new code in an interface? Or perhaps just add a `// TODO(hclog)` to it to revisit between landing on master and cutting the next major release.

Reply:
Good call. I'll Update that now as it's trivial.

Reply:
Actually I added the TODO for now as making it hclog now means adding std -> hclogger boiler plate to 4 or more places just as an interim until @s-christoff takes her pass at this.

Once we have `hclog` at the top level it's only a couple of lines change so lets do it then. (happy to help out!)